kbase-extension/static/kbase/js/widgets/narrative_core/kbaseNarrativeOutlinePanel.js

```diff
@@ -0,0 +1,208 @@
+                return stateStr + '|' + [item.depth, item.title, item.icon].join('|');
+            }, '');
+
+            if (stateStr !== this.drawnState) {
+                // Create root nested outline node
+                const root_node = { content: null, children: [], depth: 0 };
+                // Use a stack to nest outline nodes based on item depth
+                const stack = [root_node];
+                outline_items.forEach((item) => {
+                    while (stack[stack.length - 1].depth >= item.depth) {
+                        stack.pop();
+                    }
+                    const node = {
+                        item: item,
+                        children: [],
+                        depth: item.depth,
+                    };
+                    stack[stack.length - 1].children.push(node);
+                    stack.push(node);
+                });
+
+                // Render the outline tree
+                this.body.empty();
+                this.body.append(
+                    $('<div>').addClass('kb-narr-outline').append(this.renderOutlineNode(root_node))
+                );
+                this.drawnState = stateStr;
+            }
+
+            // Highlight Items
+            this.body.find('.kb-narr-outline__item').each(function (i) {
+                const item = outline_items[i];
+                $(this).toggleClass('kb-narr-outline__item--highlight', item.in_view);
+                $(this).toggleClass('kb-narr-outline__item--highlight-selected', item.selected);
+            });
+        },
+
+        /** Render a nested outline node and its children recursively */
+        renderOutlineNode: function (node) {
+            // If not the root node, create the item
+            if (node.depth !== 0) {
+                node.content = $('<div>')
+                    .addClass('kb-narr-outline__item')
+                    .attr('title', node.item.title)
+                    .append(
+                        $('<span>')
+                            .addClass('kb-narr-outline__item-icon-wrapper')
+                            .append($(node.item.icon).addClass('kb-narr-outline__item-icon'))
+                    )
+                    .append(
+                        $('<a>')
+                            .text(node.item.title)
+                            .attr('href', '#')
+                            .attr('title', node.item.title)
+                            .addClass('kb-narr-outline__item-content')
+                            .click(() => this.scrollToItem(node.item))
+                    )
+                    .click(() => this.scrollToItem(node.item));
+            }
+
+            // Render the children
+            const children = $('<ul>');
+            node.children.forEach((child) => {
+                children.append(this.renderOutlineNode(child));
+            });
+
+            if (node.depth === 0) {
+                return children;
+            } else {
```

Comment:
no need for the `else` -- there are no other options

Reply:
Its much more readable with the else IMO,